src_c/event.c

```diff
@@ -1667,9 +1667,18 @@ pg_event_post(PyObject *self, PyObject *args)
         Py_RETURN_NONE;
     }
 
-    if (pgEvent_FillUserEvent(e, &event))
-        return NULL;
+    if (e->type == SDL_KEYDOWN || e->type == SDL_KEYUP){
+        event.type =  e->type;
+        event.key.keysym.sym = PyLong_AsLong(PyDict_GetItemString(e->dict, "key"));
+        if (PyDict_GetItemString(e->dict, "scancode") != NULL)
+            event.key.keysym.scancode = PyLong_AsLong(PyDict_GetItemString(e->dict, "scancode"));
+        if (PyDict_GetItemString(e->dict, "mod") != NULL)
+            event.key.keysym.mod = PyLong_AsLong(PyDict_GetItemString(e->dict, "mod"));
```

Comment:
This line is generating the warning:
`warning C4244: '=': conversion from 'long' to 'Uint16', possible loss of data`

Reply:
What should I do here? Cast explicitly or do overflow check? Should I throw an exception if the user puts in too big numbers here?

Reply:
A cast will deal with the warning. As for what to do if the value is too big, that might be something @illume can answer. I would suggest as a minimum that a note be made in the event post() documentation that key up/down events only support 'mod' values <= 0xFFFF.